test/spec/Feature/Query/UpdateSpec.hs

```diff
@@ -535,11 +541,137 @@ spec = do
         { "id": 1, "name": "updated-item" }
       , { "id": 2, "name": "item-2" }
       , { "id": 3, "name": "item-3" }
       ]|]
+
+  context "bulk updates" $ do
+    it "can update tables with simple pk" $ do
+      baseTable "bulk_update_items" "id" tblDataBeforeBulk
+      `mutatesWith`
+      requestMutation methodPatch "/bulk_update_items" [("Prefer", "params=multiple-objects")]
+        [json|[
+          { "id": 1, "name": "item-1 - 1st", "observation": "Lost item" }
+        , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+        ]|]
+      `shouldMutateInto`
+      [json|[
+        { "id": 1, "name": "item-1 - 1st", "observation": "Lost item" }
+      , { "id": 2, "name": "item-2", "observation": null }
+      , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+      ]|]
+
+    it "can update tables with composite pk" $ do
+      baseTable "bulk_update_items_cpk" "id" tblDataBeforeBulk
+      `mutatesWith`
+      requestMutation methodPatch "/bulk_update_items_cpk" [("Prefer", "params=multiple-objects")]
+        [json|[
+          { "id": 1, "name": "item-1", "observation": "Lost item" }
+        , { "id": 2, "name": "item-2", "observation": null }
+        ]|]
+      `shouldMutateInto`
+      [json|[
+        { "id": 1, "name": "item-1", "observation": "Lost item" }
+      , { "id": 2, "name": "item-2", "observation": null  }
+      , { "id": 3, "name": "item-3", "observation": null  }
+      ]|]
+
+    it "updates using both the filters from the query string and the body" $ do
+      baseTable "bulk_update_items" "id" tblDataBeforeBulk
+      `mutatesWith`
+      requestMutation methodPatch "/bulk_update_items?id=gte.2" [("Prefer", "params=multiple-objects")]
+        [json|[
+          { "id": 1, "name": "item-1 - 1st", "observation": "Lost item" }
+        , { "id": 2, "name": "item-2 - 2nd", "observation": "Damaged item" }
+        , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+        ]|]
+      `shouldMutateInto`
+      [json|[
+        { "id": 1, "name": "item-1", "observation": null }
+      , { "id": 2, "name": "item-2 - 2nd", "observation": "Damaged item" }
+      , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+      ]|]
+
+    it "updates with limit and offset taking only the first item in the json array body" $ do
+      baseTable "bulk_update_items" "id" tblDataBeforeBulk
+      `mutatesWith`
+      requestMutation methodPatch "/bulk_update_items?limit=2&offset=1&order=id"
+        [("Prefer", "params=multiple-objects")]
+        [json|[
+          { "name": "item", "observation": "Lost item" }
+        , { "name": "item-2 - 2nd", "observation": "Damaged item" }
+        , { "name": "item-3 - 3rd", "observation": null }
+        ]|]
+      `shouldMutateInto`
+      [json|[
+        { "id": 1, "name": "item-1", "observation": null }
+      , { "id": 2, "name": "item", "observation": "Lost item" }
+      , { "id": 3, "name": "item", "observation": "Lost item" }
+      ]|]
```

Comment:
A bit confused by this one. So any value on  `limit` and `offset` will have the same outcome?

Reply:
In other words, it works as a normal update (without bulk): it takes the first element in the body and updates applying limits/offsets. Now that you mention it, it could be confusing if the user expects a bulk update of some kind, maybe a custom error should be expected here, like the one where limit/offset are not allowed for `PUT`?

Reply:
Just to be clear, `limit`/`offset` are ignored right? So `limit=100&offset=400` would result on the same?

> Now that you mention it, it could be confusing if the user expects a bulk update of some kind, maybe a custom error should be expected here, like the one where limit/offset are not allowed for PUT?

Yeah, I agree. I think we can ban `limit/offset` for bulk updates.

Reply:
> Just to be clear, limit/offset are ignored right? So `limit=100&offset=400` would result on the same?

No, it does not ignore limits and the result would be different depending on the values. In other words. It works as if it was this test https://github.com/PostgREST/postgrest/blob/b05ea14122298d2b00c4f00b28efa0c36091966f/test/spec/Feature/Query/UpdateSpec.hs#L511-L521

Reply:
I implemented the Error handling for this case and I noticed that the `PATCH` and  `DELETE` with limits and without `order` in the query string throw this error:

https://github.com/PostgREST/postgrest/blob/b05ea14122298d2b00c4f00b28efa0c36091966f/src/PostgREST/ApiRequest.hs#L253

The thing is that using the `Range` header with `order` in the query string, work as if we were using `limit` and  `offset`. Also, using `Range` without `order` throws a different error that reaches the database, with this message:

```
"message": "syntax error at or near \"RETURNING\""
```

@steve-chavez Is this behavior expected? This is unrelated to this PR, btw.

Reply:
> Also, using Range without order throws a different error that reaches the database, with this message:
> "message": "syntax error at or near \"RETURNING\""

Yeah that seems like a bug. Fixing it in another PR would be great.